zip.ml

```diff
@@ -588,8 +588,16 @@ let add_entry_header ofile comment level mtime filename =
     raise(Error(ofile.of_filename, filename, "wrong compression level"));
   if String.length filename >= 0x10000 then
     raise(Error(ofile.of_filename, filename, "filename too long"));
+  if not (Filename.is_relative filename) then
+    raise(Error(ofile.of_filename, filename, "file name must not be absolute"));
```

Comment:
The spec says:
> The path stored MUST NOT contain a drive or device letter, or a leading slash.

To be completely strict (and with idea that the constraint is about the resulting .zip file itself), one should reject e.g. `C:/foo` on all systems, not just Windows. To be noted: under Linux, `zip` happily creates a zip file with an entry named `c:\foo`.